### PR TITLE
Update to core 13.4.0

### LINKED ISSUE #8030: 10.32.3 Unable to open a realm at path
### How frequently does the bug occur?

All the time

### Description

After updating the version, the xcode 14.1 bug was solved, but a new problem came out.
ps. The real machine reports an error, but the emulator is normal.

### Stacktrace & log output

```shell
Error Domain=io.realm Code=2 "Unable to open a realm at path '/var/mobile/Containers/Data/Application/77121085-407E-4813-AC7F-26400F609BC3/Documents/localEncryptRealm/jaWords/jaWords.realm': Realm file decryption failed Path:Exception backtrace:
0   Realm                               0x0000000107c70acc _ZN5realm15InvalidDatabaseC2ERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES9_ + 60
1   Realm                               0x0000000107c1b26c _ZN5realm9SlabAlloc11attach_fileERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEERNS0_6ConfigE + 3520
2   Realm                               0x0000000107c527bc _ZN5realm2DB4openERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEbNS_9DBOptionsE + 1688
3   Realm                               0x0000000107c55818 _ZN5realm2DB4openERNS_11ReplicationERKNSt3__112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEENS_9DBOptionsE + 124
4   Realm                               0x0000000107c59f04 _ZN5realm2DB6createENSt3__110unique_ptrINS_11ReplicationENS1_14default_deleteIS3_EEEERKNS1_12basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEENS_9DBOptionsE + 156
5   Realm                               0x0000000107e858b4 _ZN5realm5_impl16RealmCoordinator7open_dbEv + 1216
6   Realm                               0x0000000107e86728 _ZN5realm5_impl16RealmCoordinator12do_get_realmENS_11RealmConfigERNSt3__110shared_ptrINS_5RealmEEENS3_8optionalINS_9VersionIDEEERNS_4util17CheckedUniqueLockE + 72
7   Realm                               0x0000000107e86594 _ZN5realm5_impl16RealmCoordinator9get_realmENS_11RealmConfigENSt3__18optionalINS_9VersionIDEEE + 380
8   Realm                               0x0000000107f14fb4 _ZN5realm5Realm16get_shared_realmENS_11RealmConfigE + 120
9   Realm                               0x0000000107b7ed5c +[RLMRealm realmWithConfiguration:queue:error:] + 728
10  RealmSwift                          0x0000000106aeb0f8 $sSo8RLMRealmC13configuration5queueABSo0A13ConfigurationC_So012OS_dispatch_C0CSgtKcfCTO + 128
11  RealmSwift                          0x0000000106b95e90 $s10RealmSwift0A0V13configuration5queueA2C13ConfigurationV_So012OS_dispatch_D0CSgtKcfC + 188
12  TomatoLMD                           0x0000000104c5b0bc $s9TomatoLMD14RealmDBHanlderC09tryToCopycF4Path33_7F36E5A86EEA9C64D2DF2892515B776DLL7dbModelyAA0C7DBModelC_tFZ + 1816
13  TomatoLMD                           0x0000000104c59a00 $s9TomatoLMD14RealmDBHanlderC10registerDB7dBModelyAA0C7DBModelC_tFZ + 640
14  TomatoLMD                           0x0000000104cde288 $s9TomatoLMD18RealmDbInitializerC04copycD00F8FromPath6dbType7classes10encryptKeyySSSg_AA0cdJ0OSaySo0C11SwiftObjectCmGSgAItFZ + 356
15  TomatoLMD                           0x0000000104cddd10 $s9TomatoLMD18RealmDbInitializerC22copyLocalFileToSandboxyyFZ + 3496
16  TomatoLMD                           0x0000000104e13bb0 $s9TomatoLMD11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtF + 460
17  TomatoLMD                           0x0000000104e154bc $s9TomatoLMD11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtFTo + 188
18  UIKitCore                           0x00000001ad628f40 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3518272
19  UIKitCore                           0x00000001ad628664 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3516004
20  UIKitCore                           0x00000001ad627640 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3511872
21  UIKitCore                           0x00000001ad62728c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3510924
22  UIKitCore                           0x00000001ad36cd64 _UIScenePerformActionsWithLifecycleActionMask + 108
23  UIKitCore                           0x00000001ad6aec3c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 4066364
24  UIKitCore                           0x00000001ad55f888 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2693256
25  UIKitCore                           0x00000001ad55f6b0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2692784
26  UIKitCore                           0x00000001ad55f260 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2691680
27  UIKitCore                           0x00000001ad55f12c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2691372
28  UIKitCore                           0x00000001adc3d5a0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 9893280
29  UIKitCore                           0x00000001adcd596c _UISceneSettingsDiffActionPerformChangesWithTransitionContextAndCompletion + 224
30  UIKitCore                           0x00000001ad40b2d8 7B942FA4-CB76-3375-9972-F58C14492FB4 + 1299160
31  UIKitCore                           0x00000001ad87d048 7B942FA4-CB76-3375-9972-F58C14492FB4 + 5959752
32  UIKitCore                           0x00000001ad4ddbec 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2161644
33  UIKitCore                           0x00000001ad4dda5c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2161244
34  UIKitCore                           0x00000001ad4dcfb0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2158512
35  UIKitCore                           0x00000001ad4dcd3c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2157884
36  FrontBoardServices                  0x00000001c0ab5d48 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 11592
37  FrontBoardServices                  0x00000001c0af5104 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 270596
38  FrontBoardServices                  0x00000001c0ab9ae4 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 27364
39  FrontBoardServices                  0x00000001c0af4d3c 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 269628
40  libdispatch.dylib                   0x00000001067a604c _dispatch_client_callout + 20
41  libdispatch.dylib                   0x00000001067a9ac8 _dispatch_block_invoke_direct + 368
42  FrontBoardServices                  0x00000001c0ac3f2c 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 69420
43  FrontBoardServices                  0x00000001c0ac3ac8 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 68296
44  FrontBoardServices                  0x00000001c0ac62a8 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 78504
45  CoreFoundation                      0x00000001ab21622c 42C5C917-0447-3995-B50F-DE4D132C2435 + 877100
46  CoreFoundation                      0x00000001ab222614 42C5C917-0447-3995-B50F-DE4D132C2435 + 927252
47  CoreFoundation                      0x00000001ab1a657c 42C5C917-0447-3995-B50F-DE4D132C2435 + 419196
48  CoreFoundation                      0x00000001ab1bbeb8 42C5C917-0447-3995-B50F-DE4D132C2435 + 507576
49  CoreFoundation                      0x00000001ab1c11e4 CFRunLoopRunSpecific + 612
50  GraphicsServices                    0x00000001e3fe1368 GSEventRunModal + 164
51  UIKitCore                           0x00000001ad670d88 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3812744
52  UIKitCore                           0x00000001ad6709ec UIApplicationMain + 340
53  libswiftUIKit.dylib                 0x00000001b3b202a0 $s5UIKit17UIApplicationMainys5Int32VAD_SpySpys4Int8VGGSgSSSgAJtF + 104
54  TomatoLMD                           0x0000000104e15ac4 $sSo21UIApplicationDelegateP5UIKitE4mainyyFZ + 104
55  TomatoLMD                           0x0000000104e15a4c $s9TomatoLMD11AppDelegateC5$mainyyFZ + 44
56  TomatoLMD                           0x0000000104e17b44 main + 28
57  dyld                                0x00000001c94e5948 341BBF64-6034-357E-8AA6-E1E4B988E03C + 88392." UserInfo={Error Code=2, NSFilePath=/var/mobile/Containers/Data/Application/77121085-407E-4813-AC7F-26400F609BC3/Documents/localEncryptRealm/jaWords/jaWords.realm, Underlying=Realm file decryption failed Path: /var/mobile/Containers/Data/Application/77121085-407E-4813-AC7F-26400F609BC3/Documents/localEncryptRealm/jaWords/jaWords.realm
Exception backtrace:
0   Realm                               0x0000000107c70acc _ZN5realm15InvalidDatabaseC2ERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES9_ + 60
1   Realm                               0x0000000107c1b26c _ZN5realm9SlabAlloc11attach_fileERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEERNS0_6ConfigE + 3520
2   Realm                               0x0000000107c527bc _ZN5realm2DB4openERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEbNS_9DBOptionsE + 1688
3   Realm                               0x0000000107c55818 _ZN5realm2DB4openERNS_11ReplicationERKNSt3__112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEENS_9DBOptionsE + 124
4   Realm                               0x0000000107c59f04 _ZN5realm2DB6createENSt3__110unique_ptrINS_11ReplicationENS1_14default_deleteIS3_EEEERKNS1_12basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEENS_9DBOptionsE + 156
5   Realm                               0x0000000107e858b4 _ZN5realm5_impl16RealmCoordinator7open_dbEv + 1216
6   Realm                               0x0000000107e86728 _ZN5realm5_impl16RealmCoordinator12do_get_realmENS_11RealmConfigERNSt3__110shared_ptrINS_5RealmEEENS3_8optionalINS_9VersionIDEEERNS_4util17CheckedUniqueLockE + 72
7   Realm                               0x0000000107e86594 _ZN5realm5_impl16RealmCoordinator9get_realmENS_11RealmConfigENSt3__18optionalINS_9VersionIDEEE + 380
8   Realm                               0x0000000107f14fb4 _ZN5realm5Realm16get_shared_realmENS_11RealmConfigE + 120
9   Realm                               0x0000000107b7ed5c +[RLMRealm realmWithConfiguration:queue:error:] + 728
10  RealmSwift                          0x0000000106aeb0f8 $sSo8RLMRealmC13configuration5queueABSo0A13ConfigurationC_So012OS_dispatch_C0CSgtKcfCTO + 128
11  RealmSwift                          0x0000000106b95e90 $s10RealmSwift0A0V13configuration5queueA2C13ConfigurationV_So012OS_dispatch_D0CSgtKcfC + 188
12  TomatoLMD                           0x0000000104c5b0bc $s9TomatoLMD14RealmDBHanlderC09tryToCopycF4Path33_7F36E5A86EEA9C64D2DF2892515B776DLL7dbModelyAA0C7DBModelC_tFZ + 1816
13  TomatoLMD                           0x0000000104c59a00 $s9TomatoLMD14RealmDBHanlderC10registerDB7dBModelyAA0C7DBModelC_tFZ + 640
14  TomatoLMD                           0x0000000104cde288 $s9TomatoLMD18RealmDbInitializerC04copycD00F8FromPath6dbType7classes10encryptKeyySSSg_AA0cdJ0OSaySo0C11SwiftObjectCmGSgAItFZ + 356
15  TomatoLMD                           0x0000000104cddd10 $s9TomatoLMD18RealmDbInitializerC22copyLocalFileToSandboxyyFZ + 3496
16  TomatoLMD                           0x0000000104e13bb0 $s9TomatoLMD11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtF + 460
17  TomatoLMD                           0x0000000104e154bc $s9TomatoLMD11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtFTo + 188
18  UIKitCore                           0x00000001ad628f40 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3518272
19  UIKitCore                           0x00000001ad628664 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3516004
20  UIKitCore                           0x00000001ad627640 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3511872
21  UIKitCore                           0x00000001ad62728c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3510924
22  UIKitCore                           0x00000001ad36cd64 _UIScenePerformActionsWithLifecycleActionMask + 108
23  UIKitCore                           0x00000001ad6aec3c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 4066364
24  UIKitCore                           0x00000001ad55f888 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2693256
25  UIKitCore                           0x00000001ad55f6b0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2692784
26  UIKitCore                           0x00000001ad55f260 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2691680
27  UIKitCore                           0x00000001ad55f12c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2691372
28  UIKitCore                           0x00000001adc3d5a0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 9893280
29  UIKitCore                           0x00000001adcd596c _UISceneSettingsDiffActionPerformChangesWithTransitionContextAndCompletion + 224
30  UIKitCore                           0x00000001ad40b2d8 7B942FA4-CB76-3375-9972-F58C14492FB4 + 1299160
31  UIKitCore                           0x00000001ad87d048 7B942FA4-CB76-3375-9972-F58C14492FB4 + 5959752
32  UIKitCore                           0x00000001ad4ddbec 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2161644
33  UIKitCore                           0x00000001ad4dda5c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2161244
34  UIKitCore                           0x00000001ad4dcfb0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2158512
35  UIKitCore                           0x00000001ad4dcd3c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2157884
36  FrontBoardServices                  0x00000001c0ab5d48 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 11592
37  FrontBoardServices                  0x00000001c0af5104 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 270596
38  FrontBoardServices                  0x00000001c0ab9ae4 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 27364
39  FrontBoardServices                  0x00000001c0af4d3c 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 269628
40  libdispatch.dylib                   0x00000001067a604c _dispatch_client_callout + 20
41  libdispatch.dylib                   0x00000001067a9ac8 _dispatch_block_invoke_direct + 368
42  FrontBoardServices                  0x00000001c0ac3f2c 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 69420
43  FrontBoardServices                  0x00000001c0ac3ac8 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 68296
44  FrontBoardServices                  0x00000001c0ac62a8 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 78504
45  CoreFoundation                      0x00000001ab21622c 42C5C917-0447-3995-B50F-DE4D132C2435 + 877100
46  CoreFoundation                      0x00000001ab222614 42C5C917-0447-3995-B50F-DE4D132C2435 + 927252
47  CoreFoundation                      0x00000001ab1a657c 42C5C917-0447-3995-B50F-DE4D132C2435 + 419196
48  CoreFoundation                      0x00000001ab1bbeb8 42C5C917-0447-3995-B50F-DE4D132C2435 + 507576
49  CoreFoundation                      0x00000001ab1c11e4 CFRunLoopRunSpecific + 612
50  GraphicsServices                    0x00000001e3fe1368 GSEventRunModal + 164
51  UIKitCore                           0x00000001ad670d88 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3812744
52  UIKitCore                           0x00000001ad6709ec UIApplicationMain + 340
53  libswiftUIKit.dylib                 0x00000001b3b202a0 $s5UIKit17UIApplicationMainys5Int32VAD_SpySpys4Int8VGGSgSSSgAJtF + 104
54  TomatoLMD                           0x0000000104e15ac4 $sSo21UIApplicationDelegateP5UIKitE4mainyyFZ + 104
55  TomatoLMD                           0x0000000104e15a4c $s9TomatoLMD11AppDelegateC5$mainyyFZ + 44
56  TomatoLMD                           0x0000000104e17b44 main + 28
57  dyld                                0x00000001c94e5948 341BBF64-6034-357E-8AA6-E1E4B988E03C + 88392, NSLocalizedDescription=Unable to open a realm at path '/var/mobile/Containers/Data/Application/77121085-407E-4813-AC7F-26400F609BC3/Documents/localEncryptRealm/jaWords/jaWords.realm': Realm file decryption failed Path:Exception backtrace:
0   Realm                               0x0000000107c70acc _ZN5realm15InvalidDatabaseC2ERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEES9_ + 60
1   Realm                               0x0000000107c1b26c _ZN5realm9SlabAlloc11attach_fileERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEERNS0_6ConfigE + 3520
2   Realm                               0x0000000107c527bc _ZN5realm2DB4openERKNSt3__112basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEEbNS_9DBOptionsE + 1688
3   Realm                               0x0000000107c55818 _ZN5realm2DB4openERNS_11ReplicationERKNSt3__112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEENS_9DBOptionsE + 124
4   Realm                               0x0000000107c59f04 _ZN5realm2DB6createENSt3__110unique_ptrINS_11ReplicationENS1_14default_deleteIS3_EEEERKNS1_12basic_stringIcNS1_11char_traitsIcEENS1_9allocatorIcEEEENS_9DBOptionsE + 156
5   Realm                               0x0000000107e858b4 _ZN5realm5_impl16RealmCoordinator7open_dbEv + 1216
6   Realm                               0x0000000107e86728 _ZN5realm5_impl16RealmCoordinator12do_get_realmENS_11RealmConfigERNSt3__110shared_ptrINS_5RealmEEENS3_8optionalINS_9VersionIDEEERNS_4util17CheckedUniqueLockE + 72
7   Realm                               0x0000000107e86594 _ZN5realm5_impl16RealmCoordinator9get_realmENS_11RealmConfigENSt3__18optionalINS_9VersionIDEEE + 380
8   Realm                               0x0000000107f14fb4 _ZN5realm5Realm16get_shared_realmENS_11RealmConfigE + 120
9   Realm                               0x0000000107b7ed5c +[RLMRealm realmWithConfiguration:queue:error:] + 728
10  RealmSwift                          0x0000000106aeb0f8 $sSo8RLMRealmC13configuration5queueABSo0A13ConfigurationC_So012OS_dispatch_C0CSgtKcfCTO + 128
11  RealmSwift                          0x0000000106b95e90 $s10RealmSwift0A0V13configuration5queueA2C13ConfigurationV_So012OS_dispatch_D0CSgtKcfC + 188
12  TomatoLMD                           0x0000000104c5b0bc $s9TomatoLMD14RealmDBHanlderC09tryToCopycF4Path33_7F36E5A86EEA9C64D2DF2892515B776DLL7dbModelyAA0C7DBModelC_tFZ + 1816
13  TomatoLMD                           0x0000000104c59a00 $s9TomatoLMD14RealmDBHanlderC10registerDB7dBModelyAA0C7DBModelC_tFZ + 640
14  TomatoLMD                           0x0000000104cde288 $s9TomatoLMD18RealmDbInitializerC04copycD00F8FromPath6dbType7classes10encryptKeyySSSg_AA0cdJ0OSaySo0C11SwiftObjectCmGSgAItFZ + 356
15  TomatoLMD                           0x0000000104cddd10 $s9TomatoLMD18RealmDbInitializerC22copyLocalFileToSandboxyyFZ + 3496
16  TomatoLMD                           0x0000000104e13bb0 $s9TomatoLMD11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtF + 460
17  TomatoLMD                           0x0000000104e154bc $s9TomatoLMD11AppDelegateC11application_29didFinishLaunchingWithOptionsSbSo13UIApplicationC_SDySo0k6LaunchJ3KeyaypGSgtFTo + 188
18  UIKitCore                           0x00000001ad628f40 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3518272
19  UIKitCore                           0x00000001ad628664 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3516004
20  UIKitCore                           0x00000001ad627640 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3511872
21  UIKitCore                           0x00000001ad62728c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3510924
22  UIKitCore                           0x00000001ad36cd64 _UIScenePerformActionsWithLifecycleActionMask + 108
23  UIKitCore                           0x00000001ad6aec3c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 4066364
24  UIKitCore                           0x00000001ad55f888 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2693256
25  UIKitCore                           0x00000001ad55f6b0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2692784
26  UIKitCore                           0x00000001ad55f260 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2691680
27  UIKitCore                           0x00000001ad55f12c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2691372
28  UIKitCore                           0x00000001adc3d5a0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 9893280
29  UIKitCore                           0x00000001adcd596c _UISceneSettingsDiffActionPerformChangesWithTransitionContextAndCompletion + 224
30  UIKitCore                           0x00000001ad40b2d8 7B942FA4-CB76-3375-9972-F58C14492FB4 + 1299160
31  UIKitCore                           0x00000001ad87d048 7B942FA4-CB76-3375-9972-F58C14492FB4 + 5959752
32  UIKitCore                           0x00000001ad4ddbec 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2161644
33  UIKitCore                           0x00000001ad4dda5c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2161244
34  UIKitCore                           0x00000001ad4dcfb0 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2158512
35  UIKitCore                           0x00000001ad4dcd3c 7B942FA4-CB76-3375-9972-F58C14492FB4 + 2157884
36  FrontBoardServices                  0x00000001c0ab5d48 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 11592
37  FrontBoardServices                  0x00000001c0af5104 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 270596
38  FrontBoardServices                  0x00000001c0ab9ae4 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 27364
39  FrontBoardServices                  0x00000001c0af4d3c 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 269628
40  libdispatch.dylib                   0x00000001067a604c _dispatch_client_callout + 20
41  libdispatch.dylib                   0x00000001067a9ac8 _dispatch_block_invoke_direct + 368
42  FrontBoardServices                  0x00000001c0ac3f2c 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 69420
43  FrontBoardServices                  0x00000001c0ac3ac8 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 68296
44  FrontBoardServices                  0x00000001c0ac62a8 8E15B35E-A614-3142-B34D-2C3CBBC13A8B + 78504
45  CoreFoundation                      0x00000001ab21622c 42C5C917-0447-3995-B50F-DE4D132C2435 + 877100
46  CoreFoundation                      0x00000001ab222614 42C5C917-0447-3995-B50F-DE4D132C2435 + 927252
47  CoreFoundation                      0x00000001ab1a657c 42C5C917-0447-3995-B50F-DE4D132C2435 + 419196
48  CoreFoundation                      0x00000001ab1bbeb8 42C5C917-0447-3995-B50F-DE4D132C2435 + 507576
49  CoreFoundation                      0x00000001ab1c11e4 CFRunLoopRunSpecific + 612
50  GraphicsServices                    0x00000001e3fe1368 GSEventRunModal + 164
51  UIKitCore                           0x00000001ad670d88 7B942FA4-CB76-3375-9972-F58C14492FB4 + 3812744
52  UIKitCore                           0x00000001ad6709ec UIApplicationMain + 340
53  libswiftUIKit.dylib                 0x00000001b3b202a0 $s5UIKit17UIApplicationMainys5Int32VAD_SpySpys4Int8VGGSgSSSgAJtF + 104
54  TomatoLMD                           0x0000000104e15ac4 $sSo21UIApplicationDelegateP5UIKitE4mainyyFZ + 104
55  TomatoLMD                           0x0000000104e15a4c $s9TomatoLMD11AppDelegateC5$mainyyFZ + 44
56  TomatoLMD                           0x0000000104e17b44 main + 28
57  dyld                                0x00000001c94e5948 341BBF64-6034-357E-8AA6-E1E4B988E03C + 88392.}
```

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

_No response_

### Version

10.32.3

### What SDK flavour are you using?

MongoDB Realm (i.e. Sync, auth, functions)

### Are you using encryption?

Yes, using encryption

### Platform OS and version(s)

iOS

### Build environment

Xcode version: ...
Dependency manager and version: ...

### PR DESCRIPTION
**Pull request by dianaafanador3**
Fixes #8030.

### CODE REVIEW

**Comment by tgoyne** [comment]
This ended up being a bunch of stuff.

Dictionary order has changed and we have a bunch of tests which check indices on unsorted dictionaries, so those tests had to be updated.

Core now exposes different semantics for an empty keypath array and a null keypath array for notifications, with null having the behavior that empty used to have. The new empty behavior is something useful that would be nice to expose, but there wasn't an obvious way to do it as a non-breaking change, so for now we just maintain the old behavior.

Core now has SyncSession::suspend() and resume() which work better than calling log_out() (now renamed to force_close()), but adopting them in a non-breaking way requires a bit of care so I didn't do it here.

The `.threadSafeReference()` combine publishers used to use a frozen Realm to pin the source version to ensure it could be resolved later. This no longer works as frozen Realms no longer pin the transaction log and live Realms can't be created at a version which only has frozen references. To fix this I introduced a new type which just holds onto a live transaction that's never advanced and hold that in the TSR instead. I also deleted a lot of unnecessarily copied and pasted code for projections and made them just use the Object publishers.

The subscribeOn() tests did some async work and relied on that completing before we managed to complete the first write transaction on the main thread. These tests were already occasionally failing on CI, but the core changes made the async task slightly slower so it started failing more often. I added some hooks so that we can take advantage of that BindingContext::before_notify() happens to be called after a notifier has been fully initialized (and at a bunch of other times, so this only works for very constrained tests).

testPassThreadSafeReferenceToDeletedObject didn't do anything to pin the source version of the TSR, so it tested something that previously happened to work but didn't work in general (e.g. performing a second write on the main thread before resolving the TSR would have made it fail). I adjusted the test to test something which is actually supposed to work.

The client reset tests need to explicitly refresh after receiving the notification that the reset has completed as that notification arrives on a background thread and can happen before the autorefresh happens. This was a pre-existing race condition in the test. While I was poking those tests I adopted XCTUnwrap where applicable.

**Review by ericjordanmossman — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,23 +1,72 @@
...
+* Pinning old versions (either with frozen Realms or with Realms on background
+  threads that simply don't get refreshed) now only prevents overwriting the
+  data needed by that version, rather than the data needed by that version and
+  all later versions. In addition, frozen Realms no longer pin the transaction
+  logs used to drive change notifications. This mostly eliminates the file size
+  growth caused by pinning versions. ([Core PR #5440](https://github.com/realm/realm-core/pull/5440))
+* Rework how Dictionaries/Maps are stored in the Realm file. The new design uses
+  less space and is typically significantly faster. This changes the iteration
+  order of Maps, so any code relying on that may be broken. We continue
+  to make no guarantees about iteration order on Maps.
```
Comment (leemaguire) [inline CHANGELOG.md:19]: Is there a link to this PR?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,23 +1,72 @@
...
+* Improve performance of acquiring read locks when a read lock for that version
+  is already held. This speeds up many operations related to change
+  notifications, and particularly refreshing a Realm which has change notifiers
+  registered.
```
Comment (leemaguire) [inline CHANGELOG.md:23]: Same as above

**Review by leemaguire — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (tgoyne) [reply CHANGELOG.md:23]: This actually shouldn't be in the changelog since it's fixing a regression we didn't ship.

**Review by dianaafanador3 — COMMENTED** [review]
Thanks for the all the hard work on this!!, I cannot approve this but I approve it by word and left some minor comments.

File: Realm/RLMRealmUtil.mm [hunk 3]
```diff
@@ -99,6 +99,16 @@ void RLMClearRealmCache() {
 public:
     RLMNotificationHelper(RLMRealm *realm) : _realm(realm) { }
 
+    void before_notify() override {
```
[the remark this answers is not in the document]
Comment (tgoyne) [reply Realm/RLMRealmUtil.mm:102]: This is purely a change to our tests.

**Pull request merged**